src/CockroachDbConnector.php

```diff
@@ -7,4 +7,18 @@
 
 class CockroachDbConnector extends PostgresConnector implements ConnectorInterface
 {
+    /**
+     * Usually the normal PostgresConnector would suffice for Cockroach,
+     * but Cockroach Serverless Clusters need an extra parameter `options`.
+     */
+    protected function getDsn(array $config)
+    {
+        $dsn = parent::getDsn($config);
+
+        if (isset($config['cluster']) && ! empty($config['cluster'])) {
+            $dsn .= ";options='--cluster={$config['cluster']}'";
+        }
```

Comment:
I have a feeling that because the options are encapsulated with single quotes we also need to add slashes. There shouldn't be any characters that need slashes in a cluster value but it would be better to have the value for the options to still be parsed and throw an error that way instead of throwing one because the encapsulation is broken.

It also might be a good idea to break this out into its own method. E.g.

```php
protected function getDsn(array $config)
{
    return $this->addClusterOption(parent::getDsn($config), $config);
}

protected function addClusterOption(string $dsn, array $config): string
{
    if (isset($config['cluster']) && ! empty($config['cluster'])) {

        return $dsn .= ";options='--cluster={$config['cluster']}'";
    }

   return $dsn;
}
```

Reply:
I followed the pattern from Laravel PostgresConnection, also, at first, I tested it using urlencode, but it failed with Cockroach claiming it wouldn't recognize the cluster, then after removing urlencode and going straight with the value at the parameter, it worked like a charm.

I also think that's exactly why Laravel changed the way they escape the database name in the DSN, in version 8 they were going directly with `database=abc`, and now they use `database='abc'`

So, what are our options here?

Reply:
have you just tried using [`addslashes`](https://www.php.net/manual/en/function.addslashes.php)?

Reply:
I added `addslashes` to the code, sadly I can't test this behavior as all my serverless clusters identification don't contain any weird chars, but with the tests passed the same way